File: pilha/estatica/Java/Pilha Com Filas/br/edu/ifba/vdc/bsi/ed/Empilhavel.java
package br.edu.ifba.vdc.bsi.ed;

public interface Empilhavel {
	void empilhar(Object elemento);
	Object desempilhar();
	Object espiar();
	void atualizar(Object dado);
	
	boolean estaCheia();
	boolean estaVazia();
	String imprimir();
}

File: pilha/estatica/Kotlin/Pilha/Empilhavel.kt
//https://www.javainuse.com/java2kot
//https://play.kotlinlang.org/
interface Empilhavel {
    // Metodos principais
    fun empilhar(dado: Any?)    //C
    fun atualizar(dado: Any?)   //U
    fun espiar(): Any?          //R
    fun desempilhar(): Any?     //D
    // Metodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}